extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -250,43 +292,31 @@ private boolean shouldAutoRedirect(TenantConfigContext configContext, RoutingCon
                 });
     }
 
-    private Uni<SecurityIdentity> performCodeFlow(IdentityProviderManager identityProviderManager,
-            RoutingContext context, TenantConfigContext configContext, String code) {
+    private boolean isRedirectFromProvider(RoutingContext context, TenantConfigContext configContext) {
+        String referer = context.request().getHeader(HttpHeaders.REFERER);
+        return referer != null && referer.startsWith(configContext.provider.getMetadata().getAuthorizationUri());
```

Comment:
I'll have to check IRL if the referer URIs are the same ones we go to, IIRC there often are quite a lot of redirects before we get back.

Reply:
@FroMage Right, but the `referer` header has a single value, or have I missed your point ? I mean, this is check is not guaranteed to succeed, but it is not a big problem - since the only thing we are deciding upon here is how to fail, see my response to Pedro as well: whether to return `401` immediately or redirect the user back to the OIDC provider to sign in again